gc3libs/config.py

```diff
@@ -218,13 +218,30 @@ def __init__(self, *locations, **extra_args):
         # save the list of (valid) config files
         self.cfgfiles = []
 
+        if not cfg_dict is None:
+            self.construct_from_cfg_dict(cfg_dict)
+
         # load configuration files if any
-        if locations:
+        elif locations:
```

Comment:
Hmmm... this would make `locations` and `cfg_dict` mutually exclusive:

1. It would need to be part of the function's docstring.
2. Why? You can start with the configuration dict provided in `cfg_dict` and merge in the result of parsing config files.

Reply:
How does e324a285fb1c30bb6ee002af5254672c934cabc9 look?

Reply:
Good, thanks!